speechbrain/lobes/models/discrete/SQ-Codec.py

```diff
@@ -0,0 +1,1497 @@
+"""This lobe enables the integration of speech codec model (SQ-Codec) with scalar quantization,.
+
+SQ-Codec effectively maps the complex speech signal into a finite and compact latent space, named scalar latent space.
+
+Repository: https://github.com/yangdongchao/SimpleSpeech
+Paper: https://arxiv.org/abs/2406.02328, https://arxiv.org/abs/2408.13893
+
+Authors
+ * Pooneh Mousavi 2024
+"""
+
+import logging
+import os
+import zipfile
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchaudio
+from huggingface_hub import hf_hub_download
+from omegaconf import OmegaConf
+from torch.autograd.function import InplaceFunction
+from torch.nn.utils import remove_weight_norm, weight_norm
+
+
+def download_and_extract(repo_id, filename, save_path):
```

Comment:
This must be refactored to use the SpeechBrain fetch function. If the fetch does not cover this use case, we should make it possible. No utility functions in lobes unless it is impossible to do otherwise.

Reply:
I could use SB fetch thc function instead of hf_hub_download, however they upload zip file and after downloading it needs to be extracted... one option is to skip downloading and ask users to download and extract the ckp and config file

Reply:
I replaced hf_hub_downloa with SB fetch but still the problem of extracting exists.